langchain4j-open-ai/pom.xml

```diff
@@ -21,11 +21,35 @@
             <artifactId>langchain4j-core</artifactId>
         </dependency>
 
+        <!-- DEPENDENCY CONFLICT RESOLUTION FOR OPENAI4J (START) -->
+
+        <!-- To resolve version conflicts inside the 'openai4j' library, we are excluding the transitive -->
+        <!-- dependencies causing version divergence errors and including them explicitly. This ensures a consistent -->
+        <!-- version to be used in the project and satisfies Maven Enforcer requirements to avoid version divergence -->
+        <!-- in the project. -->
+
+        <!-- Please check whether version conflicts are gone after upgrading  this library as this will make it -->
```

Comment:
Since https://github.com/ai-for-java/openai4j is our own client (I am the owner), we could probably resolve the conflict there?

Reply:
The proposed dependency resolution in this pull request is intended as a temporary measure to enable the Maven enforcer plugin ASAP, preventing further conflicting dependencies from being added. If/when `dev.ai4j;openai4j` will be updated to a conflict-free version, the block within <!-- DEPENDENCY CONFLICT RESOLUTION FOR AZURE-AI-OPENAI (START) --> can be removed ...

Side note:
The conflict actually originates from `com.squareup.okhttp3:okhttp`. Based on this [thread ](https://github.com/square/okhttp/issues/8288) they are not going to fix it in version 4.x.x (which is being used in `dev.ai4j;openai4j`) but it's reportedly alraedy fixed in pre-release 5.x.x version.

Anyway - if you are the owner of `openai4j` library then you could definitelly move this dependency fix there...